pkg/templates/parser.go

```diff
@@ -0,0 +1,145 @@
+package templates
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/projectdiscovery/nuclei/v3/pkg/catalog"
+	"github.com/projectdiscovery/nuclei/v3/pkg/catalog/config"
+	"github.com/projectdiscovery/nuclei/v3/pkg/utils"
+	"github.com/projectdiscovery/nuclei/v3/pkg/utils/stats"
+	yamlutil "github.com/projectdiscovery/nuclei/v3/pkg/utils/yaml"
+	fileutil "github.com/projectdiscovery/utils/file"
+	"gopkg.in/yaml.v2"
+)
+
+type Parser struct {
+	ShouldValidate bool
+	NoStrictSyntax bool
+	// this cache can be copied safely between ephemeral instances
+	parsedTemplatesCache *Cache
+	// this cache might potentially contain references to heap objects
+	// it's recommended to always empty it at the end of execution
+	compiledTemplatesCache *Cache
+}
+
+func NewParser() (*Parser, error) {
+	p := &Parser{
+		parsedTemplatesCache:   NewCache(),
+		compiledTemplatesCache: NewCache(),
+	}
+
+	return p, nil
+}
```

Comment:
are we planning to do any validations or accept opts as input here in near future ? if not i think we can just remove error from return parameters ?

Reply:
Mostly I left it as a placeholder. But it seems like at current time it's not that meaningful, going to remove it shortly